src/arrayMethodSort.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction) {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - i - 1; j++) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - i - 1; j++) {
+          if (String(this[j]) > String(this[j + 1])) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        }
+      }
```

Comment:
instead of if else logic, just add the default value for compareFunction
<img width="242" alt="image" src="https://github.com/mate-academy/js_array-method-sort/assets/70220667/80670640-8f22-477c-b0d3-2e53aac23157">

Reply:
I see, will try to.